### PR TITLE
added all functions

### PR DESCRIPTION
**Pull request by mayaraz**

### CODE REVIEW

File: whose_file [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+no_space = lambda lst: [item.lstrip() for item in lst]
+
+remove_line_whitespace = lambda s: '\n'.join(map(lambda st: st.strip(), s.splitlines()))
+
+has_dup = lambda l: len([item for item in l if l.count(item) > 1]) > 0
+
+flatten = lambda l: [item for lst in l for item in lst]
+
+grep = lambda folder, string: [path for path in glob(os.path.join(folder, '*')) if string in open(path, 'r').read()]
```
Comment (egerby) [inline whose_file:16]: This won’t work if there is a directory in the path given.

Comment (egerby) [inline whose_file:16]: os also allows you to see the contents of a dir.

Comment (mayaraz) [reply whose_file:16]: I know, but I thought it will be better this way since `os.path.listdir` returns only file names and not full paths

Comment (mayaraz) [reply whose_file]: Fixed it

**Pull request merged**